bro_connector/main/management/commands/frd_sync_to_bro.py

```diff
@@ -1,13 +1,396 @@
+import os
+import time
+
 from django.core.management.base import BaseCommand
+from bro_exchange.broxml.frd.requests import FrdStartregistrationTool
+from bro_exchange.bhp.connector import (
+    validate_sourcedoc,
+    upload_sourcedocs_from_dict,
+    check_delivery_status,
+)
+from frd.models import FormationResistanceDossier, FrdSyncLog
+from datetime import datetime
+from main.settings.base import FRD_SETTINGS
+
+#############
+## Command ##
+#############
 
 
 class Command(BaseCommand):
     """
     Handles the sync between the bro-connector and the BRO.
     Runs daily as cronjob.
     """
-    help = 'Syncs the FRD data to the BRO'
+
+    help = "Syncs the FRD data to the BRO"
 
     def handle(self, *args, **kwargs):
+        self.handle_startregistrations()
+        self.handle_measurements()
+
+    def handle_startregistrations(self):
+        """
+        Handles the startregistrations for all FRD's without bro_id.
+        For each new FRD, the FrdStartregistration class is used to hande the registration
+        """
+        new_dossiers_qs = self.get_new_dossiers()
+
+        if len(new_dossiers_qs) != 0:
+            for new_dossier in new_dossiers_qs:
+                startregistration = FrdStartregistration(new_dossier)
+                startregistration.sync()
+
+    def get_new_dossiers(self):
+        return FormationResistanceDossier.objects.filter(frd_bro_id=None)
+
+    def handle_measurements(self):
+        """
+        Develop when startregistrations are done
+        """
         pass
 
+
+##############################
+## General helper functions ##
+##############################
+
+
+def update_log_obj(log_obj, field_value_dict):
+    """
+    Updates the date modified field to now, and changes the values found in the dict
+    """
+    now = datetime.now()
+    log_obj.date_modified = now.strftime("%d/%m/%Y, %H:%M:%S")
+
+    for key, value in field_value_dict.items():
+        setattr(log_obj, key, value)
+
+    log_obj.save()
+
+
+#############
+## Classes ##
+#############
+
+
+class FrdStartregistration:
```

Comment:
Dit is vast degene waar je het liefst commentaar op krijgt, maar dat laat ik heel even zitten:
- Kijk eerst even naar de rest, dat is meer standaardcommentaar.
- Deze class, da's de èchte aansturing, daar kunnen we beter even voor bij een whiteboard gaan staan.
- Ik snap de class niet echt. De reden is dat er nogal wat losse onderdelen zijn. Deze class bestuurt het process, maar slaat deels indirect via-via logs op in models die op andere plekken dan weer ingelezen worden: er zitten afhankelijkheden in die het lastig maken het te volgen. De verantwoordelijkheid lijkt wat verspreid te zijn. Daar moeten we naar kijken.